sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -298,3 +299,34 @@ case class StringLength(child: Expression) extends UnaryExpression with ExpectsI
 
   override def prettyName: String = "length"
 }
+
+/**
+ * Returns string, with the first letter of each word in uppercase,
+ * all other letters in lowercase. Words are delimited by whitespace.
+ */
+case class InitCap(child: Expression) extends UnaryExpression with ExpectsInputTypes {
+  override def dataType: DataType = StringType
+  override def inputTypes: Seq[DataType] = Seq(StringType)
+  override def eval(input: InternalRow): Any = {
+    val string = child.eval(input)
+    if (string == null) {
+      null
+    } else {
+      val str: String = string.toString
+      if (str == "") {
+        return null
+      }
+      val sb = new StringBuilder
+      sb.append(str)
+      sb.setCharAt(0, sb(0).toUpper)
+
```

Comment:
tarekauel: I think this check should be implemented in stringbuilder function. On the other side, the exception is reasonable. We cannot hide all the exceptions.